src/Auth0/Provider.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace SocialiteProviders\Auth0;
+
+use SocialiteProviders\Manager\OAuth2\User;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'AUTH0';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [
+        'openid',
+        'profile',
+        'email',
+    ];
+
+    protected $scopeSeparator = ' ';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuth0Url()
+    {
+        return $this->getConfig('base_url');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function additionalConfigKeys()
+    {
+        return ['base_url'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase($this->getAuth0Url().'/authorize', $state);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return $this->getAuth0Url().'/oauth/token';
+
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get($this->getAuth0Url().'/userinfo', [
+            'headers' => [
+                'Authorization' => 'Bearer ' . $token,
+            ],
+        ]);
+
+        return json_decode($response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'id' => $user['sub'],
+            'nickname' => $user['nickname'],
+            'name' => $user['given_name'] . ' ' . $user['family_name'],
```

Comment:
Will these always be set? Not clear from the docs.

Reply:
Everything I have seen says yes.

Reply:
You could use the null coalescing operator so that it doesn't error on accident: https://stackoverflow.com/questions/34571330/php-ternary-operator-vs-null-coalescing-operator

'name' => $user['given_name'] ?? '' . ' ' . $user['family_name'] ?? '',

I expect that it would be good to do for the other values here.

Reply:
Alternatively you can use array get `Arr::get($user, 'given_name', '')`